source/collections/Queue.ooc

```diff
@@ -12,17 +12,20 @@ Queue: abstract class <T> {
 	dequeue: abstract func -> (T, Bool)
 	peek: abstract func -> (T, Bool)
 	//TODO: This works much better and will do for now
-	dequeue: abstract func ~out (result: T*) -> Bool
+	dequeue: abstract func ~out (result: T*) -> Bool // Also crashing the compiler
```

Comment:
This sounds complicated enough to deserve an explanation in the file itself. `// Also crashing the compiler` is not helpful. Also, what do you mean by "functional"? That it works, as opposed to the existing function?